### PR TITLE
misc tweaks on golang docs

### PR DESCRIPTION
**Pull request by cblgh**
Now that I am taking my first steps into go-ssb proper (having only lived on the outskirts of it during the [go-ssb-room](https://github.com/ssb-ngi-pointer/go-ssb-room) and [netsim](https://github.com/ssb-ngi-pointer/netsim) projects) I'm tweaking the golang documentation with the eye of a first-time reader :^)

I don't really expect anyone other than @cryptix to review this (but the more the merrier, I guess!)

:black_heart:

### CODE REVIEW

File: docs/golang/README.md [hunk 1]
```diff
@@ -136,21 +136,43 @@ The `graph` package goes through all the `type:contact` messages and just stores
...
-Enter [MultiLog](https://pkg.go.dev/go.cryptoscope.co/margaret/multilog), it returns `margaret.Log`s for specific addresses.
+Enter [MultiLog](https://pkg.go.dev/go.cryptoscope.co/margaret/multilog), which returns `margaret.Log`s for specific addresses.
 
-#### Implementation details
+Package `multilog` is one of a few ways that go-ssb creates an overarching structure (indexes and views) of the
+messages being received from all over the scuttleverse. Another prominent indexing solution is
+the [`margaret/indexes`](https://godocs.io/go.cryptoscope.co/margaret/indexes) package, which
+more or less functions as a key-value store. One of the differences between `margaret/indexes` and
+`margaret/multilog`: the former stores arbitrary data (think: strings of your choosing), while
+the latter only stores ssb messages.
```
Comment (cryptix) [inline docs/golang/README.md:150]:
```suggestion
the latter only stores (references to) ssb messages.
```

File: docs/golang/README.md [hunk 2]
```diff
@@ -136,21 +136,43 @@ The `graph` package goes through all the `type:contact` messages and just stores
...
+#### A `multilog` explanation
 
-A newer approach is to store the set of receivelog sequences as a special kind of [Bitmap](https://en.wikipedia.org/wiki/Bitmap). (tl;dr: You can imagine these as a compressed array of integers.) This is advantageous not only because we only store one bitmap for all the messages in a set (like `by author:x` or `by type:y`) but also because it allows us to use them as compound indexes since they can be logically combined via boolean algebra (`x AND y` gives us the intersection, `x OR y` gives us the union) since they all map to the same messages in the receive log.
+One way to think about the `multilog` package: 
+
+A multilog can be seen as a _[tree](https://en.wikipedia.org/wiki/Tree_(data_structure))-like_
+index. The multilog itself is a log—or the root of the tree—which leads to other logs. These
+other logs are called _sublogs_, and each sublog has many leaves. Each leaf corresponds to some
+message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (all follow messages by `<peer>`, for example).
```
Comment (cryptix) [inline docs/golang/README.md:159]:
```suggestion
message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (i.e. all `type:contact` messages, all messages by `<peer>`, etc.).
```

Comment (cryptix) [reply docs/golang/README.md:159]: the example is not directly indexed but could be constructed by boolean combination of two indexes.

**Review by cryptix — APPROVED** [review]
SGTM! Thanks a lot for the clarifications.

**Comment by cblgh** [comment]
:tada:

**Pull request merged**